draft-irtf-cfrg-frost.md

```diff
@@ -559,55 +612,78 @@ signature and commitment shares using DeserializeElement for each. If validation
 fails, the Coordinator MUST abort the protocol. If validation succeeds, the
 Coordinator then verifies the set of signature shares using the following procedure.
 
-<!-- the inputs to this function need to be revisited, things can probably be made simpler -->
+## Signature Share Verification and Aggregation {#frost-aggregation}
+
+After signers perform round two and send their signature shares to the Coordinator,
+the Coordinator verifies each signature share for correctness. In particular,
+for each signer, the Coordinator uses commitment pairs generated during round
+one and the signature share generated during round two, along with other group
+parameters, to check that the signature share is correct using the following procedure.
+
 ~~~
   Inputs:
-  - group_comm, the group commitment
+  - index, Index `i` of the signer. Note index will never equal `0`.
+  - public_key_share_i, the public key for the ith signer, where public_key_share = G.ScalarBaseMult(s[i])
+  - comm_i, pair of Element values (hiding_nonce_commitment, binding_nonce_commitment) generated
+    in round one from the ith signer.
+  - sig_share_i, a Scalar value indicating the signature share as produced in round two from the ith signer.
+  - commitment_list = [(j, hiding_nonce_commitment_j, binding_nonce_commitment_j), ...], a list of commitments
+    issued by each signer, where each element in the list indicates the signer index j and their
+    two commitment Element values (hiding_nonce_commitment_j, binding_nonce_commitment_j).
+    This list MUST be sorted in ascending order by signer index.
   - participant_list, a set containing identifiers for each signer, similarly of length
     NUM_SIGNERS (sent by the Coordinator).
-  - index, Index `i` of the signer. Note index will never equal `0`.
   - group_public_key, the public key for the group
-  - public_key_share, the public key for the ith signer, where public_key_share = G.ScalarBaseMult(s[i])
-  - sig_share, the signature share for the ith signer, computed from the signer
-  - comm_share, the commitment for the ith signer, computed from the signer
   - msg, the message to be signed
 
-  Outputs: 1 if the signature share is valid, and 0 otherwise.
+  Outputs: True if the signature share is valid, and False otherwise.
 
-  def verify_signature_share(group_comm, participant_list, index, group_public_key, public_key_share, sig_share, comm_share, msg):
-    group_comm_enc = G.SerializeElement(group_comm)
-    group_public_key_enc = G.SerializeElement(group_public_key)
-    challenge_input = group_comm_enc || group_public_key_enc || msg
-    c = H2(challenge_input)
+  def verify_signature_share(index, public_key_share_i, comm_i, sig_share_i, commitment_list,
```

Comment:
It's currently used directly, rather than being extracted from the commitment list, so I'm leaving this as-is for now. In a followup change, we should go through and implement helpers for mapping the commitment list to individual commitment values and participant lists.